scripts/tunnel-creation.mjs

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env node
+
+import { logger } from '@appium/support';
+import net from 'node:net';
+
+import {
+  PacketStreamServer,
+  TunnelManager,
+  createLockdownServiceByUDID,
+  createUsbmux,
+  startCoreDeviceProxy,
+  startTunnelRegistryServer,
+} from 'appium-ios-remotexpc';
+
+const log = logger.getLogger('TunnelCreationTest');
+
+/**
+ * Update tunnel registry with new tunnel information
+ * @param {Array} results - Array of tunnel results
+ * @returns {Object} Updated tunnel registry
+ */
+async function updateTunnelRegistry(results) {
+  const now = Date.now();
+  const nowISOString = new Date().toISOString();
+
+  const registry = {
+    tunnels: {},
+    metadata: {
+      lastUpdated: nowISOString,
+      totalTunnels: 0,
+      activeTunnels: 0,
+    }
+  };
+
+  for (const result of results) {
+    if (result.success) {
+      const udid = result.device.Properties.SerialNumber;
+      registry.tunnels[udid] = {
+        udid,
+        deviceId: result.device.DeviceID,
+        address: result.tunnel.Address,
+        rsdPort: result.tunnel.RsdPort ?? 0,
+        packetStreamPort: result.packetStreamPort,
+        connectionType: result.device.Properties.ConnectionType,
+        productId: result.device.Properties.ProductID,
+        createdAt: registry.tunnels[udid]?.createdAt ?? now,
+        lastUpdated: now,
+      };
+    }
+  }
+
+  registry.metadata = {
+    lastUpdated: nowISOString,
+    totalTunnels: Object.keys(registry.tunnels).length,
+    activeTunnels: Object.keys(registry.tunnels).length, // Assuming all are active for now
+  };
+
+  return registry;
+}
+
+const activeServers = [];
+const packetStreamServers = new Map();
+
+const deviceInfoMap = new Map();
+
+const INFO_SERVER_PORT = 49152;
+let PACKET_STREAM_BASE_PORT = 50000;
+
+/**
+ * Create or get the info server that provides information about all devices
+ * @returns {Promise<Object>} Server info object with server and port
+ */
+async function getInfoServer() {
```

Comment:
maybe it would make sense to group similar methods into classes rather than have them all in the global scope

Reply:
@mykola-mokhnach this classes your mentioning should it come from remotexpc repo or just move these commands into a class within the scripts directory?